pkg/flow/reconciler/xslttransform/reconciler.go

```diff
@@ -35,17 +37,26 @@ type reconciler struct {
 
 	// Knative Service reconciler
 	ksvcr libreconciler.KServiceReconciler
+
+	sinkResolver *resolver.URIResolver
 }
 
 // Check that our Reconciler implements Interface
 var _ reconcilerv1alpha1.Interface = (*reconciler)(nil)
 
 // ReconcileKind implements Interface.ReconcileKind.
 func (r *reconciler) ReconcileKind(ctx context.Context, o *v1alpha1.XSLTTransform) pkgreconciler.Event {
-	o.Status.InitializeConditions()
-	o.Status.ObservedGeneration = o.Generation
+	var url string
```

Comment:
Why not passing the strongly typed `*apis.URL` returned by the resolver, untouched? It would allow you to simply compare against `nil`.